desktopexporter/internal/server/server_test.go

```diff
@@ -0,0 +1,321 @@
+package server
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/CtrlSpice/otel-desktop-viewer/desktopexporter/internal/telemetry"
+	"github.com/stretchr/testify/assert"
+)
+
+func setupWithTrace(t *testing.T) (*Server, func(*testing.T)) {
+	s := NewServer("localhost:8000")
+	testSpanData := telemetry.SpanData{
+		TraceID:      "1234567890",
+		TraceState:   "",
+		SpanID:       "12345",
+		ParentSpanID: "",
+		Name:         "test",
+		Kind:         "",
+		StartTime:    time.Now(),
+		EndTime:      time.Now().Add(time.Second),
+		Attributes:   map[string]interface{}{},
+		Events:       []telemetry.EventData{},
+		Links:        []telemetry.LinkData{},
+		Resource:     &telemetry.ResourceData{Attributes: map[string]any{"service.name": "pumpkin.pie"}, DroppedAttributesCount: 0},
+		Scope: &telemetry.ScopeData{
+			Name:                   "test.scope",
+			Version:                "1",
+			Attributes:             map[string]any{},
+			DroppedAttributesCount: 0,
+		},
+		DroppedAttributesCount: 0,
+		DroppedEventsCount:     0,
+		DroppedLinksCount:      0,
+		StatusCode:             "",
+		StatusMessage:          "",
+	}
+
+	if err := s.Store.AddSpans(context.Background(), []telemetry.SpanData{testSpanData}); err != nil {
+		t.Fatalf("could not create test span: %v", err)
+	}
+	return s, func(t *testing.T) {
+		s.Store.Close()
+	}
+}
+func TestTracesHandler(t *testing.T) {
+	t.Run("Traces Handler (Empty)", func(t *testing.T) {
+		s := NewServer("localhost:8000")
+		defer s.Store.Close()
+
+		testSummaries := telemetry.TraceSummaries{
+			TraceSummaries: []telemetry.TraceSummary{
+				{
+					HasRootSpan:     true,
+					RootServiceName: "groot",
+					RootName:        "i.am.groot",
+					RootStartTime:   time.Now(),
+					RootEndTime:     time.Now().Add(time.Minute),
+					SpanCount:       2,
+					TraceID:         "12345",
+				},
+			},
+		}
+
+		req, err := http.NewRequest("GET", "/api/traces", nil)
+		if err != nil {
+			t.Fatalf("could not create request: %v", err)
+		}
+
+		rec := httptest.NewRecorder()
+		s.tracesHandler(rec, req)
+		res := rec.Result()
+		defer res.Body.Close()
+
+		b, err := io.ReadAll(res.Body)
+		if err != nil {
+			t.Fatalf("could not read response: %v", err)
+		}
+
+		if res.StatusCode != http.StatusOK {
```

Comment:
Not required, but the `assert` library can be used for these equality assertions:

```go
assert.Equal(t, http.StatusOK, res.StatusCode)
```